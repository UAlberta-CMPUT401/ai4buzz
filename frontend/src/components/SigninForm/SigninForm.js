import React from "react";

const SigninForm = ({
  authType,
  onSubmit,
  onEmailChange,
  onPasswordChange,
}) => {
  const handleSubmit = (event) => {
    event.preventDefault();
    onSubmit();
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        placeholder="email"
        onChange={(e) => onEmailChange(e.target.value)}
      />
      <input
        placeholder="password"
        onChange={(e) => onPasswordChange(e.target.value)}
      />
      <button>{authType}</button>
    </form>
  );
};

export default SigninForm;
