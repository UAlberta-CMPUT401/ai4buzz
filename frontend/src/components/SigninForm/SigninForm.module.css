.signinForm {
  width: 25%;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid rgba(0, 0, 0, 0.1);
  border-radius: 1rem;
  padding: 1rem;
  box-shadow: rgba(0, 0, 0, 0.25) 0px 14px 28px,
    rgba(0, 0, 0, 0.22) 0px 10px 10px;
  background-color: #121826;
}

@media only screen and (max-width: 600px) {
  .signinForm {
    width: 95%;
    box-shadow: none;
    background-color: transparent;
  }
}

.signinInputEmail,
.signinInputPassword {
  display: block;
  font-size: 1.12em;
  padding: 0.4rem;
  border-radius: 0.5rem;
  border: 0;
  background: #262d3c;
  outline: none;
  margin-bottom: 1rem;
  color: #fff;
  width: 100%;
}

.signinInput::placeholder {
  color: #888e99;
}

.submitButton {
  display: block;
  font-size: 1.1rem;
  padding: 0.4rem 1.5rem;
  border-radius: 0.5rem;
  cursor: pointer;
  color: #253251;
  border: none;
}

.submitButton:disabled {
  background-color: gray;
}

.labelName {
  color: #888e99;
  font-size: 0.9rem;
}

.switchAuth {
  color: #fff !important;
}
