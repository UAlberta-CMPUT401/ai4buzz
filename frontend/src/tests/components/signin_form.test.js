import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SigninForm from '../../components/SigninForm/SigninForm';

it("Renders SigninForm for auth type 'SignUp' correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SigninForm
          authType="SignUp"
          onSubmit={() => {}}
          onEmailChange={() => {}}
          onPasswordChange={() => {}}
          error=""
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Renders SigninForm for auth type 'Login' correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SigninForm
          authType="Login"
          onSubmit={() => {}}
          onEmailChange={() => {}}
          onPasswordChange={() => {}}
          error=""
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
