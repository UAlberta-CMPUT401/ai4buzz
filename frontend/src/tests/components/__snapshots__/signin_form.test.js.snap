// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SigninForm for auth type 'Login' correctly 1`] = `
<form
  className="signinForm"
  onSubmit={[Function]}
>
  <h1>
    Login
  </h1>
  <label
    className="labelName"
  >
    Email
  </label>
  <input
    className="signinInputEmail"
    id="signin__email__input"
    onChange={[Function]}
    placeholder="email"
    type="email"
  />
  <label
    className="labelName"
  >
    Password
  </label>
  <input
    className="signinInputPassword"
    id="signin__password__input"
    onChange={[Function]}
    placeholder="password"
    type="password"
  />
  <div
    className="buttonGradient"
  >
    <button
      className="submitButton"
      disabled={false}
    >
      Login
    </button>
  </div>
  <a
    className="switchAuth"
    href="/signup"
    onClick={[Function]}
  >
    Signup
  </a>
</form>
`;

exports[`Renders SigninForm for auth type 'SignUp' correctly 1`] = `
<form
  className="signinForm"
  onSubmit={[Function]}
>
  <h1>
    SignUp
  </h1>
  <label
    className="labelName"
  >
    Email
  </label>
  <input
    className="signinInputEmail"
    id="signin__email__input"
    onChange={[Function]}
    placeholder="email"
    type="email"
  />
  <label
    className="labelName"
  >
    Password
  </label>
  <input
    className="signinInputPassword"
    id="signin__password__input"
    onChange={[Function]}
    placeholder="password"
    type="password"
  />
  <div
    className="buttonGradient"
  >
    <button
      className="submitButton"
      disabled={false}
    >
      SignUp
    </button>
  </div>
  <a
    className="switchAuth"
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
</form>
`;
